### ISSUE TITLE
[java]  RedundantFieldInitializer FN with constant expressions

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3684]
**Affects PMD Version:6.41.0** 

**Rule:RedundantFieldInitializer**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#redundantfieldinitializer>

**Description:**
Hi, I found a false negative about the rule RedundantFieldInitializer. 

**Code Sample demonstrating the issue:**

```java
class Impl {
    int a = 1 + 0 - 1; // should report a warning
    int b = 0 + 1 - 1; // can be detected
}
```
**Expected outcome:**

PMD should report a violation at line 2, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by dykov** [comment]
note: 
in your example the first literal in the second field declaration is 0.
```java
int b = 0 + 1 - 1; // can be detected
```
the rule searches only first `PrimaryPrefix/Literal` node here: https://github.com/pmd/pmd/blob/4ae7e40bad8a1c1a9bfada21c3608b80e1c616a8/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/performance/RedundantFieldInitializerRule.java#L106
so the 3rd row is detected, not because the result is 0

**Comment by oowekyala** [comment]
I implemented logic to compute constant expressions in PMD 7. This was a pain to write and test, and investing time to do it in PMD 6 would be a waste of time. The rule already supports any constant expressions in pmd 7.

As @dykov pointed out, `0 + 1 - 1` is only flagged by chance. Expressions starting with 0 but not evaluating to zero (eg `0+1`) would cause a false positive which should be fixed in pmd 6.

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.
